Employee_Management_Portal/EmployeeManagementPortal/EmployeeManagementPortal/src/main/java/com/portal/service/AdminService.java

```diff
@@ -0,0 +1,183 @@
+            .getLogger(AdminService.class);
+
+    /**
+     * Decodes Base64 encoded data.
+     *
+     * @param encodedData The encoded data to decode.
+     * @return The decoded data as a string.
+     */
+    public static String decodeData(final String encodedData) {
+        byte[] decodedBytes = Base64.getDecoder().decode(encodedData);
+        return new String(decodedBytes, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Registers a new admin.
+     *
+     * @param adminDTO The AdminDTO containing admin details.
+     * @return The registered Admin as an AdminDTO.
+     * @throws DuplicateEntryException if an admin with the same email already
+     *                                 exists.
+     */
+    public final ResponseDTO registerAdmin(final EmployeeInDTO adminDTO) {
+        Employee adminEntity = new Employee();
+
+        adminEntity.setRole(Role.ADMIN);
+        adminEntity.setManagerId(0L);
+
+        adminEntity.setName(adminDTO.getName());
+        adminEntity.setEmail(adminDTO.getEmail());
+        adminEntity.setEmpId(adminDTO.getEmpId());
+        adminEntity.setDesignation(adminDTO.getDesignation());
+        adminEntity.setContactNumber(adminDTO.getContactNumber());
+        adminEntity.setDob(adminDTO.getDob());
+        adminEntity.setDoj(adminDTO.getDoj());
+        adminEntity.setLocation(adminDTO.getLocation());
+        adminEntity.setPassword(adminDTO.getPassword());
+        adminEntity.setManagerId(adminDTO.getManagerId());
+        adminEntity.setRole(adminDTO.getRole());
+
+        this.adminRepository.save(adminEntity);
+        ResponseDTO response = new ResponseDTO();
+        response.setMessage("Admin added successfully");
+        response.setRole("ADMIN");
+
+        return response;
+    }
+
+    /**
+     * Handles admin login.
+     * @param loginUser The LoginDTO containing login credentials.
+     * @return The authenticated Admin as an AdminDTO, or null if authentication
+     *         fails.
+     * @throws ResourceNotFoundException if the username is not found.
+     */
+    public final LoginOutDTO login(final LoginInDTO loginUser) {
+       // LOGGER.error("Email does not exist");
+        Optional<Employee> employee = adminRepository
+                .findByEmail(loginUser.getEmail());
+        if (employee.isPresent() && passwordEncoder.matches(
+                decodeData(loginUser.getPassword()),
+                employee.get().getPassword())) {
+            LOGGER.info("User Logged in");
+            LoginOutDTO response = new LoginOutDTO();
+            response.setMessage("Login Successful");
+            response.setName(employee.get().getName());
+            response.setRole(employee.get().getRole());
+            response.setMessage("Login Successful");
+            response.setId(employee.get().getId());
+            return response;
+        }
+
+        LOGGER.error("Wrong Credentials");
+        throw new WrongCredentialsException("Wrong Credentials");
```

Comment:
do not pass string message directly when throwing exceptions. Make another class ('ex: ErrorConstants.java) and store all messages in that class as constants and then call it here.